apps/mobile-wallet/src/persistent-storage/wallet.ts

```diff
@@ -87,15 +85,51 @@ export const generateAndStoreWallet = async (
   }
 }
 
+const generateAndStoreWalletMetadata = async (name: WalletStoredState['name'], isMnemonicBackedUp: boolean) => {
+  const firstAddressHash = await generateAndStoreAddressKeypairForIndex(0)
+  const walletMetadata = generateWalletMetadata(name, firstAddressHash, isMnemonicBackedUp)
+  await storeWalletMetadata(walletMetadata)
+
+  return {
+    id: walletMetadata.id,
+    firstAddressHash
+  }
+}
+
 export const getWalletMetadata = async (): Promise<WalletMetadata | null> => {
-  const rawWalletMetadata = await getWithReportableError(WALLET_METADATA_STORAGE_KEY)
+  let walletMetadata = null
 
   try {
-    return rawWalletMetadata ? JSON.parse(rawWalletMetadata) : null
+    walletMetadata = await getParsedRawWalletMetadata()
   } catch (error) {
     sendAnalytics({ type: 'error', error, message: 'Could not parse wallet metadata' })
-    throw error
   }
+
+  // Fallback mechanism when metadata is not found
+  if (!walletMetadata && (await storedMnemonicV2Exists())) {
+    await generateAndStoreWalletMetadata('My wallet', false)
+
+    try {
+      walletMetadata = await getParsedRawWalletMetadata()
+    } catch (error) {
+      sendAnalytics({ type: 'error', error, message: 'Could not parse recreated wallet metadata' })
+      throw error
+    }
+
+    if (walletMetadata) {
+      sendAnalytics({ type: 'error', message: 'Re-created missing wallet metadata' })
+    } else {
+      sendAnalytics({ type: 'error', message: 'Could not re-created missing wallet metadata' })
+    }
+  }
+
+  return walletMetadata
+}
```

Comment:
If we detect a stored mnemonic but no metadata, we regenerate with some default values (`My wallet`, not backed up)